dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/message/LengthFieldStreamingDecoder.java

```diff
@@ -130,16 +127,12 @@ private void deliver() {
     }
 
     private void processHeader() throws IOException {
-        ByteArrayOutputStream bos = new ByteArrayOutputStream(lengthFieldOffset + lengthFieldLength);
         byte[] offsetData = new byte[lengthFieldOffset];
         int ignore = accumulate.read(offsetData);
-        bos.write(offsetData);
         processOffset(new ByteArrayInputStream(offsetData), lengthFieldOffset);
         byte[] lengthBytes = new byte[lengthFieldLength];
         ignore = accumulate.read(lengthBytes);
-        bos.write(lengthBytes);
         requiredLength = bytesToInt(lengthBytes);
-        this.dataHeader = new ByteArrayInputStream(bos.toByteArray());
 
         // Continue reading the frame body.
         state = DecodeState.PAYLOAD;
```

Comment:
**Issue**: The removal of 'ByteArrayOutputStream' and direct processing of byte arrays from the 'InputStream' is a significant change that could impact how message headers are handled. Ensure that this change does not affect the protocol's expected behavior, especially in edge cases with large headers or payloads. <br> **Fix**: Thoroughly test the new implementation with various message sizes and formats to ensure compatibility with the expected HTTP/2 framing and encoding behaviors. <br> **Code Suggestion**: 
 ```
 -        ByteArrayOutputStream bos = new ByteArrayOutputStream(lengthFieldOffset + lengthFieldLength);
          byte[] offsetData = new byte[lengthFieldOffset];
          int ignore = accumulate.read(offsetData);
 -        bos.write(offsetData);
          processOffset(new ByteArrayInputStream(offsetData), lengthFieldOffset);
          byte[] lengthBytes = new byte[lengthFieldLength];
          ignore = accumulate.read(lengthBytes);
 -        bos.write(lengthBytes);
          requiredLength = bytesToInt(lengthBytes);
 -        this.dataHeader = new ByteArrayInputStream(bos.toByteArray());
 ```